enhancements/sdk-metrics.md

```diff
@@ -0,0 +1,120 @@
+---
+title: sdk-metrics
+  - "@varshaprasad96"
+reviewers:
+  - "@dmesser"
+  - "@joelanford"
+approvers:
+  - "@dmesser" 
+  - "@joelanford"
+creation-date: 2020-05-19
+last-updated: 2020-05-19
+status: implementable
+---
+
+# Operator SDK metrics
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Summary
+
+The enhancement is intended to define the bundle resources in which SDK labels can be added, so that they can be read by OLM and included in the current OLM metrics. These metrics will help us in providing a measure to evaluate the reachability of Operator SDK among community operators.
+
+## Motivation
+
+Collecting metrics regarding the usage of Operator SDK among operator authors is important to measure the reachability of the tool. Since OLM already has the pipeline built to obtain metrics on the operators and report it to the redhat telemetry, we intend to utilize the same to obtain data on the operators using SDK to build bundle images.
+
+### Goals
+
+The goal of this enhancement is to :
+1.  List the bundle resources on which SDK stamp will be added.
+2. The format of the stamp which will be added.
+
+### Design details
+
+The current bundle image has the following project structure:
+
+```
+ ...
+  bundle.Dockerfile
+  ...
+  $ tree ./deploy/olm-catalog/test-operator/
+  ./deploy/olm-catalog/test-operator/
+  ├── manifests
+  │   ├── example.com_tests_crd.yaml
+  │   └── test-operator.clusterserviceversion.yaml
+  └── metadata
+      └── annotations.yaml
+```
+
+SDK stamps would be present in the following locations:
+
+1. bundle.Dockerfile
+
+The dockerfile will include the sdk stamp in the form of `LABEL` which will follow the following format:
+
+```
+LABEL operators.operatorframework.io.bundle.sdk.v1 = operator_type:sdk_version
+```
+
+2. annotations.yaml
+
+Similar to the format followed in the dockerfile, `annotations.yaml` will have the following label:
+
+```
+operators.operatorframework.io.bundle.sdk.v1: operator_type:sdk_version
+```
+
+For example, in case of memcached-operator, the labels can be written as:
+
+```
+LABEL operators.operatorframework.io.bundle.sdk.v1 = go:v0.17.0
+```
+
+3. Cluster service version
```

Comment:
Does OLM's upcoming support for CSV-less bundles impact the approach for SDK stamping?

In that future scenario, SDK will not be involved in CSV generation so we'll have no opportunity to inject anything directly into the CSV.

Will OLM have the necessary metadata to inject these annotations on behalf of SDK? If not, does it make sense to include these stamps on the CSV now only to have them removed in the near-ish future? Are stamps on other resource types sufficient?

Reply:
We don't have concrete a plan for operators that don't ultimately create a CSV on cluster, though it's likely we'll migrate our on-cluster representation to something else once CSVs are no longer shipping in bundles.

> Will OLM have the necessary metadata to inject these annotations on behalf of SDK? 

We don't have any such mechanism today (bundle -> runtime object projection), that would be net-new. We can pair on adding something like that to this enhancement?

> If not, does it make sense to include these stamps on the CSV now only to have them removed in the near-ish future?

If we want to punt on that discussion, we could write the label into the bundle and the CSV for now?

Reply:
For now we should annotate CSVs. We can do this with current and new projects because we control CSV generation.

When CSV-less bundles become a thing it would be nice to set arbitrary annotations (and, tangentially, labels) in `olm.yaml` that are applied to the internal operator representation that OLM constructs, whether that be a CSV or something else.

We should also consider annotating CRDs, in case we need to use those labels in the future. We could do this with patches for new projects; we control CRD generation for current projects.